AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -331,6 +331,40 @@ - (void)dealloc
 
 #pragma mark - Core
 
+- (void)__tearDown:(BOOL)tearDown subnodesOfNode:(ASDisplayNode *)node
+{
+  for (ASDisplayNode *subnode in node.subnodes) {
+    if (tearDown) {
+      [subnode __unloadNode];
+    } else {
+      [subnode __loadNode];
+    }
+  }
+}
+
+- (void)__unloadNode
+{
+  ASDisplayNodeAssertThreadAffinity(self);
+  ASDN::MutexLocker l(_propertyLock);
+  
+  if (_flags.layerBacked)
+    _pendingViewState = [_ASPendingState pendingViewStateFromLayer:_layer];
+  else
+    _pendingViewState = [_ASPendingState pendingViewStateFromView:_view];
+    
+  [_view removeFromSuperview];
+  _view = nil;
+  if (_flags.layerBacked)
+    _layer.delegate = nil;
+  [_layer removeFromSuperlayer];
+  _layer = nil;
+}
+
+- (void)__loadNode
+{
+  [self layer];
```

Comment:
This seems a little sparse, but smart :).  Multiple loads and unloads of a node is a first for the public ASDK, something we had some challenges implementing a while back when building prototypes of view reuse.  I assume each time will use layerBlock or viewBlock accordingly?